lib/ram_pwrdn/ram_pwrdn.c

```diff
@@ -46,26 +54,48 @@ static const struct ram_bank banks[] = {
 #else
 	{ .start = 0x20010000UL, .section_count = 6, .section_size = 0x8000 },
 #endif
+#elif defined(CONFIG_SOC_NRF5340_CPUAPP)
+	{ .start = 0x20000000UL, .section_count = 16, .section_size = 0x1000 },
+	{ .start = 0x20010000UL, .section_count = 16, .section_size = 0x1000 },
+	{ .start = 0x20020000UL, .section_count = 16, .section_size = 0x1000 },
+	{ .start = 0x20030000UL, .section_count = 16, .section_size = 0x1000 },
+	{ .start = 0x20040000UL, .section_count = 16, .section_size = 0x1000 },
+	{ .start = 0x20050000UL, .section_count = 16, .section_size = 0x1000 },
+	{ .start = 0x20060000UL, .section_count = 16, .section_size = 0x1000 },
+	{ .start = 0x20070000UL, .section_count = 16, .section_size = 0x1000 },
+#endif
 };
 
 /*
  * Power down selected RAM sections of the given bank.
  */
 static void ram_bank_power_down(uint8_t bank_id, uint8_t first_section_id, uint8_t last_section_id)
 {
+#if defined(CONFIG_SOC_NRF52840) || defined(CONFIG_SOC_NRF52833)
 	uint32_t mask = GENMASK(NRF_POWER_RAMPOWER_S0POWER + last_section_id,
 				NRF_POWER_RAMPOWER_S0POWER + first_section_id);
 	nrf_power_rampower_mask_off(NRF_POWER, bank_id, mask);
+#elif defined(CONFIG_SOC_NRF5340_CPUAPP)
+	uint32_t mask = GENMASK(VMC_RAM_POWER_S0POWER_Pos + last_section_id,
```

Comment:
How about adding following at the top of the file
#if defined(CONFIG_SOC_NRF52840) || defined(CONFIG_SOC_NRF52833)
#define RAM_POWER_REG NRF_POWER_RAMPOWER_S0POWER
#elif defined(CONFIG_SOC_NRF5340_CPUAPP)
#define RAM_POWER_REG VMC_RAM_POWER_S0POWER_Pos
#endif

and leaving this and ram_bank_power_up function without ifdefs?

Reply:
Can be merged without this change